lib/ask_web/controllers/respondent_controller.ex

```diff
@@ -1106,6 +1116,8 @@ defmodule AskWeb.RespondentController do
     project = load_project_for_owner(conn, project_id)
     survey = load_survey(project, survey_id)
 
+    channels = survey_channel_names(survey)
```

Comment:
It wasn't necessary per-se to load the list of channels _once_, but it allows to spare a preload request for each batch, leading to halve the number of SQL requests we make, which is always nice (and there are usually just a single digit number of channels, so we don't hog memory).

Reply:
We could probably spare preloading the `questionnaire` for each respondent batch in `results` too: we already load the questionnaires for the survey, and have available.

Reply:
Answering myself: one results request was preloading the respondent questionnaires but it never used it, and actually used the questionnaires loaded from the survey. I also replaced

They don't bring noticeable speed improvements (returns just a few rows, always the same, already cached in MySQL), but it spares loooots of repeated requests.